internal/devolucion.go

```diff
@@ -0,0 +1,23 @@
+package internal
+
+type EstadoDevolucion int
+
+const (
+	Bueno EstadoDevolucion = iota
```

Comment:
Has cambiado esto en 426582c, y lo has hecho en referencia a un comentario. Incluso los comentarios deben seguir un issue y una HU. Ahora es muy difícil saber qué es lo que estás solucionando y si realmente corresponde a una HU o a un issue creado por ti. Incluso la versión anterior 0a91a5d tampoco hace ninguna referencia a ningún issue, con lo que es muy difícil ver (tanto por tu parte como por parte de @giorgiogiovanni ) si estás añadiendo valor al cliente.

El problema, además, es que en los dos commits estás modificando muchos ficheros. Eso hace extremadamente complicado saber los cambios específicos que estás haciendo. Si no seguís la metodología desde el principio al final, la revisión se hace imposible.

Finalmente he llegado al commit original, que es este: 7787ed8

El mensaje de commit dice: 

> Modularización para mejor entendimiento de https://github.com/giorgiogiovanni/PacketManager/issues/7, https://github.com/giorgiogiovanni/PacketManager/issues/8, https://github.com/giorgiogiovanni/PacketManager/issues/9, https://github.com/giorgiogiovanni/PacketManager/issues/10 y https://github.com/giorgiogiovanni/PacketManager/issues/11

Hay varios problemas en ese código
1. Estás cambiando 6 ficheros y borrando 1
2. Estás refiriéndote prácticamente a todos los issues. Es imposible por tanto saber en qué estás trabajando de forma precisa, qué modelización del problema has hecho y por qué. Has creado un montón de issues porque yo he dicho que creéis un montón de issues y luego un montón de código porque parece que teníais que crear un montón de código. No has creado issues porque hayas analizado el problema y llegado finalmente a una fracción del mismo fácilmente abordable, no, porque *no crees* que crear issues sirva para nada. Si no entiendes por qué se debe hacer algo, simplemente no lo hagas. Haz lo que entiendas y sigue a partir de ahí. Para la persona que revisa, pedir una revisión de todo y sobre todo revisar todo es mucho más complicado y menos eficiente que ir haciendo cambios incrementales, poco a poco. En eso se basa el desarrollo ágil: mejoras incrementales.
3. De esos 6 ficheros, has dejado un `main`. Una vez más, has escrito código porque hay que hacer código, o le has pedido a ChatGPT que lo haga. No has seguido la metodología, que era mirar qué producto mínimamente viable (con énfasis en mínimamente) pedía. El [milestone 0](https://github.com/giorgiogiovanni/PacketManager/milestone/1) dice literalmente "Código que represente los elementos del dominio." Una función `main`, ¿Qué elemento del dominio es? Una función *cualquiera*, ¿es un elemento del dominio?

Todo esto viene a contestar el comentario que has hecho. Evidentemente no tenéis experiencia en este tipo de cosas. Pero ¿tenéis experiencia en *creer* en lo que estáis haciendo? ¿Que si  habéis hecho milestones en el objetivo 1, es para que los uséis en el objetivo 2? ¿Que aplicar una metodología resulta en trabajo más productivo y más eficiente? Finalmente, ¿que el profesor pide cosas no porque le apetezca o porque tenga esa manía, sino porque realmente *el si cree* que lo que trata de que aprendáis es para que podáis ser ingenieros del software más productivos y eficientes?

También viene a decir que sin un rastro que lleve del código al issue y de este a la HU este código es inevaluable. Puede que el estado de la devolución sea binario, ternario, necesario o innecesario. Soy incapaz de decir si es válido o no. Puedo fiarme de @giorgiogiovanni y que luego él se busque la vida en su objetivo 4, pero en ese caso tú no habrás aprendido nada, porque la lección más importante no es sepáis escribir mejor o peor código, es simplemente que si no se sigue la metodología el código es inválido.

Reply:
En cuanto a esto es verdad que en un momento determinado cuando le solicité que reflexionara en cuestiones semánticas y modulara, es cierto que hizo un gran trabajo de cambios.. pero me limité a ver el estado final del código  no a trackear por commit-issue (error por mi parte, pero eran muchos archivos y cambios)

Respecto al estado del código creo que es válido para seguir trabajando. @JJ, se me plantea una duda cuando comenta que _es simplemente que si no se sigue la metodología el código es inválido_. Llegados a este punto la pregunta _¿Qué hago ahora?_ creo que nos la planteamos tanto @antoniorr02 como yo.

Reply:
Pues hay que mirar los issues de nuevo y programar desde cero, de forma que el blame del código pueda ver claramente qué es lo que implementa y por qué

Reply:
Perfecto, cancelamos entonces esta PR, para iniciar desde una rama _from scratch_ y que se puedan trackear los cambios. ¿De acuerdo, @antoniorr02?

Reply:
En respuesta a @JJ me queda la duda de que si un revisor te hace una propuesta de cambio que afecta a varios ficheros (como por ejemplo eliminar todos los comentarios del codigo o la mutabilidad) y no te asocia ningún issue, en ese caso, ¿tengo que deducir yo a que issue asociarlo, crear un nuevo issue para el commit, o proponer los nuevos cambios en base al comentario del revisor?  Porque si esto último cumple con la metodología no entiendo el inconveniente de lo que he hecho, pues lo veo más eficiente que ir haciendo un commit por cada pequeña modificación en cada fichero asociando un issue (o incluso creando nuevos issues) descomponiendo cada proposición de cambio. No veo entonces porque se entiende por esto metodología de desarrollo ágil.

Por otro lado, lo que comenta del archivo main, si sigue la trazabilidad de mis commit podrá ver que de primeras, realicé muchas funciones para el posterior uso del modelado, pensando que estas eran parte del modelo. Con ello, cree un main para comprobar que estas funcionaban correctamente. Si bien, con las recomendaciones de @giorgiogiovanni  (de gran ayuda por cierto), este ha quedado obsoleto y de hecho ni existe en la versión actual.

Reply:
En cualquier caso no tengo problema en comenzar de 0 con otro PR como propone @giorgiogiovanni, pero antes me gustaría aclarar todos estos aspectos para aprender de ellos y no volver a cometer los mismos fallos de forma recurrente.

Reply:
Los cambios deben ser siempre atómicos y siempre referirse a un issue. Si no, otro revisor (o sea, yo), puede verlos y no darse cuenta de qué van...